src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -66,50 +118,136 @@ class DefaultCheckbox extends Component<CheckboxProps> {
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
+          ].join(' '),
+        };
+      }
+      return {};
+    };
+
+    const newCheckboxInputProps = {
+      disabled,
+      id,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      'aria-invalid': status === 'error',
+      checked: !!checkedState,
+      className: checkboxClassNames.input,
+      onChange: this.handleClick,
+      name,
+      ...getDescribedBy(),
+      ...(value ? { value } : {}),
+    };
+
     return (
-      <StyledCheckbox
-        disabled={disabled}
-        onClick={this.handleClick}
-        {...passProps}
-        status={status}
-        className={classnames(baseClassName, className, {
-          [checkboxClassNames.error]: status === 'error' && !disabled,
-          [checkboxClassNames.checked]: checkedState && !disabled,
-          [checkboxClassNames.large]: variant === 'large',
-          [checkboxClassNames.disabled]: !!disabled,
-        })}
-      >
-        {!!checkedState && (
-          <Icon
-            icon="check"
-            className={classnames(iconBaseClassName, {
-              [iconClassnames.checked]: checkedState && !disabled,
-              [iconClassnames.error]: status === 'error' && !disabled,
-              [iconClassnames.disabled]: !!disabled,
-            })}
-          />
+      <HtmlDiv
+        className={classnames(
+          checkboxClassNames.container,
+          className,
+          baseClassName,
+          {
+            [checkboxClassNames.error]: status === 'error' && !disabled,
+            [checkboxClassNames.checked]: checkedState && !disabled,
+            [checkboxClassNames.large]: variant === 'large',
+            [checkboxClassNames.disabled]: !!disabled,
+          },
         )}
-        {children}
-      </StyledCheckbox>
+      >
+        <HtmlInput {...newCheckboxInputProps} type="checkbox" />
+        <HtmlLabel
+          htmlFor={id}
+          className={checkboxClassNames.label}
+          {...passProps}
+        >
+          {!!checkedState && (
+            <Icon
+              icon="check"
+              className={classnames(iconBaseClassName, {
+                [iconClassnames.checked]: checkedState && !disabled,
+                [iconClassnames.error]: status === 'error' && !disabled,
+                [iconClassnames.disabled]: !!disabled,
+              })}
+            />
+          )}
+          {children}
+        </HtmlLabel>
+        <HintText id={hintTextId}>{hintText}</HintText>
+        <StatusText id={statusTextId} status={status}>
+          {statusText}
+        </StatusText>
+      </HtmlDiv>
     );
   }
 }
 
+const StyledCheckbox = styled(
+  ({
+    tokens,
+    id: propId,
+    ...passProps
+  }: CheckboxProps & InternalTokensProp) => (
+    <AutoId id={propId}>
+      {(id) => <BaseCheckbox id={id} {...passProps} />}
+    </AutoId>
```

Comment:
Added the use of AutoId here instead of the exporting of the class. 
Now it only needs to be told once. In export part it would have been be told twice; for render and for the static.